Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,286 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Frames
```

Comment:
As a rule of thumb, class names should be nouns that are generally singular; but should describe the purpose of the class. 

`Frame` is a fine name for a class, but I would expect `Frames` to be a variable name describing a collection of `Frame` objects. To further this, I would expect a class named `Frame` to be a plain old data object having no action methods.

Try to think of a name that describes what this class does; if it's hard to think of such a name that may be an indication that this class too many responsibilities (then see if you can split it up into things you can name).

Reply:
I'd like to see you write some unit tests for the public functions in this class, that may help clear up some of the other problems I see in this file.

Start by creating a new test class to house all tests for this production class. Then one by one write test cases for each possible input for each public function, and verify it does what you expect.

Some tools that could help you in this endeavor are:
  * [AutoFixture](https://github.com/AutoFixture/AutoFixture) - Creates random data for use in tests. This can be useful especially here for invalid input test cases; or alternatively test cases where you need to generate some data that you don't care what the value is.
  * [AutoMoq](https://github.com/darrencauthon/AutoMoq) - Creates mock objects to stand-in for interfaces. When/If you split out new classes from this class, it can be helpful to create an interface for the new class' public methods. This will give you a well defined interface to code to, removes the implementation knowledge of those details from this class, and makes it super easy to mock. When you mock an interface in a unit test you make sure you are only testing a single class (in this case it would be the `Frames` class). So if a test fails, you can be assured the failure is in `Frames` and not in the classes it calls. The short and dirty version of mocking is: create a fake object that adheres to the given interface which you can setup to return whatever value you want given specific inputs.

When it comes time to test the methods that are currently calling `Console.WriteLine`, I would suggest using the [Proxy Pattern](https://www.dofactory.com/net/proxy-design-pattern) to create a proxy that calls `Console.WriteLine` directly. This will give you the control to make an interface for your proxy, and allow you to mock the interface in your tests. That will allow you to verify that you are writing the correct things to the console, while decoupling this class from the type of console it is writing to.

I know some of this can be hard to grok at first, and this is a large ask that will be convoluted at times; feel free to ask any of us to pair with you or answer any questions you may have.

Reply:
I am thinking that BowlingScore may work for this rename.  I could then move any step by step logic from main() and add those as functions in BowlingScore.